lib/matplotlib/contour.py

```diff
@@ -1117,15 +1117,20 @@ def _autolev(self, N):
 
         return lev[i0:i1]
 
-    def _process_contour_level_args(self, args):
+    def _process_contour_level_args(self, args, z_dtype):
         """
         Determine the contour levels and store in self.levels.
         """
         if self.levels is None:
-            if len(args) == 0:
-                levels_arg = 7  # Default, hard-wired.
-            else:
+            if args:
                 levels_arg = args[0]
+            elif z_dtype.char == "?":  # bool
```

Comment:
Why not

```suggestion
            elif z_dtype == np.bool:
```
?

Or `np.issubdtype` if you want to support subclasses.

Reply:
Should that be `np.bool_`?
https://numpy.org/doc/stable/user/basics.types.html

Reply:
Thanks, fixed.  (Just "bool" works OK here, np.issubdtype handles it as needed.)